plugin/internal/remotecall/remotecall.go

```diff
@@ -0,0 +1,235 @@
+package remotecall
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+const (
+	UserAgent           = "User-Agent"
+	UserAgentSSHGateway = "SSH-Gateway"
+	Accept              = "Accept"
+	ContentType         = "Content-Type"
+	ApplicationJson     = "application/json"
+
+	AuthTokenUserClusterMapping = "authToken"
+	Authorization               = "Authorization"
+	IdentitySSHGateway          = "SSH-gateway"
+	JwtValidity                 = 5 * time.Second
+)
+
+type RemoteCall struct {
+	userClusterNameURL *url.URL
+	userClusterToken   string
+
+	clusterNameInClusterAuthenticatorURL map[string]*url.URL
```

Comment:
can we have better (shorter) name :)

Reply:
i am not able to think of something :(

Reply:
`clusterNameToAuthenticatorURL` should be good enought?

Reply:
userClusterURLFormat - i think i will skip this - fixed other two